src/contracts/atlas/Atlas.sol

```diff
@@ -209,21 +208,22 @@ contract Atlas is Escrow, Factory {
         for (uint256 i; i < solverOpsLength; ++i) {
             _bidAmountFound = _getBidAmount(ctx, dConfig, userOp, solverOps[i], returnData);
 
-            if (_bidAmountFound == 0 || _bidAmountFound > type(uint240).max) {
-                // Zero bids are ignored: increment _zeroBidCount offset
-                // Bids that would cause an overflow are also ignored
-                unchecked {
-                    ++_zeroBidCount;
-                }
-            } else {
+            // skip zero and overflow bid's
+            if (_bidAmountFound != 0 && _bidAmountFound <= type(uint240).max) {
                 // Non-zero bids are packed with their original solverOps index.
-                // The array is filled with non-zero bids from the right. This causes all zero bids to be on the left -
-                // in their sorted position, so fewer operations are needed in the sorting step below.
-                _bidsAndIndices[_bidsAndIndicesLastIndex - (i - _zeroBidCount)] =
-                    uint256(_bidAmountFound << _BITS_FOR_INDEX | uint16(i));
+                // The array is filled with non-zero bids from the right.
+                _bidsAndIndices[_bidsAndIndicesLastIndex] = uint256(_bidAmountFound << _BITS_FOR_INDEX | uint16(i));
+            }
+
+            // decrease for each iteration
+            unchecked {
+                --_bidsAndIndicesLastIndex;
             }
         }
```

Comment:
```suggestion
                // The array is filled with non-zero bids from the right.
                // All zero bids are then implicitly moved to the left. This makes insertion sort below more efficient.
                _bidsAndIndices[_bidsAndIndicesLastIndex] = uint256(_bidAmountFound << _BITS_FOR_INDEX | uint16(i));

                // decrease the pointer for each non-zero bid placed from the right
                unchecked {
                    --_bidsAndIndicesLastIndex;
                }
            }
        }
```
I think this decrement of the pointer should only happen when we place a non-zero bid on the right. Otherwise we may end up with zero bids in between the non-zero bids. We want all the zero bids on the left of the array to make the `insertionSort()` operation as efficient as possible.